### DISCUSSION TITLE
RPC unexpectedly clearing out entire table

### DISCUSSION THREAD

**Discussion by taylorhlabs** [supabase/supabase#3586] [Questions] [answered]
I have a simple function (parameter: resource_id uuid; language: sql; return type: void) that marks an item as deleted and then deletes any matching records from a link table.
```
update resources as r set deleted = true where r.id = resource_id;
delete from boards_resources as br where br.resource_id = resource_id;
``` 
From the front end, I call the function using the `rpc` API:
```
const { data, error } = await supabase.rpc(
  'delete_resource',
  { resource_id: resourceId }
);
```
In the `resources` table, the resource with matching id is correctly marked as deleted. But for some reason, the entire `boards_resources` table is cleared out instead of just deleting the record(s) with matching `resource_id`. Any ideas why the function is clearing out the entire `boards_resources` table?

**Comment by soedirgo** [comment, marked as answer]
Can you prefix the parameter name to `_resource_id`? Presumably `resource_id` in the second line refers to the table's column instead of the parameter, so it deletes the entire table (because the condition is a tautology).

**Reply by taylorhlabs** [reply]
That did the trick! Thanks @soedirgo
